RELEASE.txt updates

lrknox opened the pull request.

lrknox committed:
Update platforms tested details

lrknox committed:
Merge branch 'hdf5_1_14_6' of https://github.com/HDFGroup/hdf5 into update_platforms_tested

lrknox committed:
Merge branch 'hdf5_1_14_6' of https://github.com/HDFGroup/hdf5 into update_platforms_tested

lrknox committed:
Add HL_FORTRAN_f90_tstlite failure with cce/18.0.1 to known problems.
Minor text modifications to clarify fixes.

lrknox committed:
Remove unused NEWSLETTER files.

byrnHDF commented on release_docs/NEWSLETTER.txt:
This file is used by the Release Workflows.

byrnHDF reviewed (changes requested):
Restore The NEWSLETTER file with the majoe items to be noted. Needed for release.

lrknox committed:
Revert "Remove unused NEWSLETTER files."
NEWSLETTER.txt still needed for github.com/HDFGroup/hdf5/releases.

This reverts commit 4ab050d6f3c75089dc112b3a6176c27ec97173df.

lrknox committed:
Update test in release_docs/NEWSLETTER.txt for release page description.

lrknox committed:
Corrected line in NEWSLETTER.txt for release page.
Added 3 Fortran tests that would fail with cce/18.0.1 if they weren't
skipped.  Removed tests that are no longer failing.
